common-rendering/src/components/LastUpdated.tsx

```diff
@@ -6,19 +6,22 @@ const LastUpdated = ({
 	lastUpdated,
 }: {
 	lastUpdatedDisplay: string;
-	lastUpdated: number;
+	lastUpdated: Date;
 }) => {
 	return (
 		<div
-			css={css`
+		css={css`
 				display: flex;
 				align-items: flex-end;
 				${textSans.xxsmall()};
 				color: ${neutral[46]};
-			`}
+				`}
 		>
-			<time dateTime={new Date(lastUpdated).toISOString()}>
-				{`Updated at ${lastUpdatedDisplay}`}
+			<time
+				className="js-last-updated-time"
```

Comment:
I know you and Jamie are happy to go forward with the `js-*` prefix for AR but this code is in shared and will end up in DCR and in that case this is not a good idea. DCR took the line that we would not use this pattern.

The reason we made that call is to avoid confusion. We know that Frontend widely makes use of 'js-' selectors but DCR has none, except where we need to support legacy code.

In fact, in DCR we don't use selectors hardly at all. We use React and select content using the React scope. In those situations where we do need to select elements outside of React we create contracts, which, so far, follow the pattern of using either data attributes like `data-relativeformat` or pseudo custom elements, like `gu-hydrate`

If we merge this code then we breach that line and open up the possibility for a very loose contract for how elements are selected which I really do think is a bad idea for DCR.

Reply:
Why don't we extend the _existing_ `time` contract that is already in place by having `data-absoluteformat: 'local' | 'edition'`? That way it works for both AR and DCR, which both treat time differently.

We already have `data-relativeformat: 'med' | 'long'` and it feels like keeping these `time` related contracts consistent is a good thing

Reply:
Thanks @oliverlloyd - I can see how this could work within DCR. I'm hoping @JamieB-gu might be able to comment from an AR-perspective as this approach would be a new pattern within that codebase (i.e. selecting using `data-relativeformat` instead of className `js-last-updated-time`).

I'm wondering whether, if we can't reach an agreement, this component shouldn't be shared (maybe that's too extreme)?

Reply:
> I'm wondering whether, if we can't reach an agreement, this component shouldn't be shared (maybe that's too extreme)?

I don't think that's an extreme take. Trying to make shared code work when the use cases are different leads to complex, hard to maintain abstractions. Perhaps if we can't find a pattern that works cross platform here then it makes more sense to prioritise the benefits of consistent patterns and future maintainability vs. the benefits of having shared code in this case?

Reply:
> open up the possibility for a very loose contract for how elements are selected which I really do think is a bad idea for DCR.

My suggestion would be to use one or the other, _not_ to use both. I agree that would be unhelpful 👍

> DCR took the line that we would not use this pattern.
> 
> The reason we made that call is to avoid confusion. We know that Frontend widely makes use of 'js-' selectors but DCR has none, except where we need to support legacy code.

Does this mean that `data-*` attributes were chosen over `js-*` classes primarily to distinguish DCR code from frontend code? Are there other reasons for preferring them?

Just to make sure I understand: we're talking about two different conventions for how elements that are rendered on the server-side can be selected (looked up) on the client-side? One approach is to select based on `data-*` attributes, the other is to select based on `js-*` classes? Both ultimately achieve the same goal, so it's a question of pros/cons of each?

Assuming I have understood correctly (shout if I haven't), I'd argue that `js-*` classes are preferable because this is the only thing they're used for. If you see a `js-*` class you _know_ it's being used to select code on the client-side somewhere. Data attributes, by contrast, are part of a broader web API, and therefore may be used for other purposes as well. They're also designed to be key-value pairs, so that you can transfer information from the HTML context to the JS context. It seems strange to use them just to look up the element, without caring about the data in the attribute, but it may well be that I've not understood the context correctly.

Anyway, all that said, I think the benefit of consistency and being able to share a convention outweighs the above advantages. As mentioned, I think we pick one and run with it. My preference would be to shift DCR over to `js-*` classes, but if you think that would be undesirable/problematic then we can shift AR over to `data-*` attributes instead and go with that.

Reply:
We made the call to only use classnames for styling and use attributes for building contracts in DCR. There's not really any kind of agreement on which way to go and both are valid but my feeling is data attributes are more scalable. Being able to have a structured key pair arrangement is a lot easier to scale than a single string.

Some of the oldest contracts in place at the Guardian use data attributes (`data-component`, `data-link-name`, `data-relativeformat`, etc.) and I - personally - think these might not have endured so long had they been classnames. 

This is [the discussion](https://github.com/guardian/dotcom-rendering/pull/2669) introducing the `data-ignore` contract. There are a lot of others including contracts with other teams (`data-spacefinder`) I worry trying to change all these would be a lot of work and add a lot of risk.

Reply:
Fair enough 👍. In which case I think it's reasonable to switch to `data-*` attributes in AR @DavidLawes. There's clearly been some thought behind this in DCR, and I think we get a lot of benefit from sharing this convention and the resulting code 📈